### DISCUSSION TITLE
What about to have structs in GDScript?

### DISCUSSION THREAD

**Discussion by azur-wolve** [godotengine/godot-proposals#4670] [Scripting]
GDScript already has _**inner classes**_, but they are passed by value or by reference?
Having `struct`s passed by value supposes boost → memory cache optimization.
And are great to pass multiple values between functions either by passing as args or returning them, cheap in resources.

As far as I know _inner classes_ only are meant to be used within the Type that holds them, for code organization. 
But can't be passed in between Types, unless both hold the same inner class (both would be supposed to be identical).

It would make sense or there's something that I'm missing here?

**Comment by Calinou** [comment]
This is already being tracked in https://github.com/godotengine/godot-proposals/issues/2816.
